tls/s2n_psk.c

```diff
@@ -26,12 +26,12 @@
 
 #define S2N_HASH_ALG_COUNT S2N_HASH_SENTINEL
 
-int s2n_psk_init(struct s2n_psk *psk, s2n_psk_type type)
+int s2n_psk_init(struct s2n_psk *psk, s2n_psk_type type, s2n_hmac_algorithm hmac_alg)
 {
     notnull_check(psk);
 
     memset_check(psk, 0, sizeof(struct s2n_psk));
-    psk->hmac_alg = S2N_HMAC_SHA256;
+    psk->hmac_alg = hmac_alg;
```

Comment:
This is incorrect as it may set the hmac_alg to ` S2N_HMAC_NONE`. The default hash algorithm MUST be set to `S2N_HMAC_SHA256`. From RFC https://tools.ietf.org/html/rfc8446#section-4.2.11:

```
   For externally established PSKs, the Hash algorithm MUST be set when
   the PSK is established or default to SHA-256 if no such algorithm is
   defined.  The server MUST ensure that it selects a compatible PSK
   (if any) and cipher suite.
   ````

Reply:
You could handle it by checking if the hmac_alg passed in is NONE and setting it to 256, but as you noticed when updating all those tests, using a default and just requiring hmac_alg to be set later if necessary is pretty convenient :)